### PR TITLE
Implement duplicate properties warning

### LINKED ISSUE #27: Validate application.properties: duplicate properties
Given this application.properties:

```
greeting.message=A
greeting.message=B
```

greeting.message should be marked as warn because properties are duplicated.

CodeAction could remove a key

### PR DESCRIPTION
**Pull request by xorye**
Fixes #27 

In this PR, only the duplicate property(s) have a warning range:
image
 
@angelozerr , I'm not sure if this is what you meant storing the property in a set. Please let me know if I should change anything.

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java [hunk 1]
```diff
@@ -40,12 +42,14 @@
...
 
 	private final QuarkusValidationSettings validationSettings;
+	private final Set<String> propertyNameSet;
 
 	public QuarkusValidator(QuarkusProjectInfo projectInfo, List<Diagnostic> diagnostics,
 			QuarkusValidationSettings validationSettings) {
 		this.projectInfo = projectInfo;
 		this.diagnostics = diagnostics;
 		this.validationSettings = validationSettings;
+		this.propertyNameSet = new HashSet<String>();
```
Comment (angelozerr) [inline quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java]: rename propertyNameSet with existingProperties

Comment (xorye) [reply quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java]: I have made the changes

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java [hunk 2]
```diff
@@ -97,7 +101,13 @@ private void validateDuplicateProperty(String propertyName, Property property) {
 			// The duplicate validation must be ignored for this property name
 			return;
 		}
-		// TODO : store in the set the property to know if it already exsists
+
+		if (existingProperties.contains(propertyName)) {
+			addDiagnostic("Duplicate property '" + propertyName + "'", property.getKey(), severity,
```
[the remark this answers is not in the document]
Comment (xorye) [reply quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java]: I have updated the PR.

image

**Comment by angelozerr** [comment]
Great job @xorye ! Could you please create a PR to add this validation settings in vscode-quarkus in package.json.

**Comment by xorye** [comment]
Thanks, @angelozerr! Is the validation setting supposed to be for severity?

**Comment by angelozerr** [comment]
> Thanks, @angelozerr!

You are welcome!

> Is the validation setting supposed to be for severity?

IMHO I think we should have severity settings for each validation kind.

**Pull request merged**